lib/liquid/combinators/tag.ex

```diff
@@ -59,4 +84,32 @@ defmodule Liquid.Combinators.Tag do
     |> ignore(string("end" <> tag_name))
     |> parsec(:end_tag)
   end
+
+  def store_tag_in_context(_rest, [{:block, tag}] = acc, %{tags: tags} = context, _line, _offset) do
+    tag_name = tag |> Keyword.keys() |> hd() |> to_string()
+    {acc, %{context | tags: [tag_name | tags]}}
+  end
+
+  def check_allowed_tags(_, acc, %{tags: []} = context, _, _, _) do
```

Comment:
What are the first, fourth, and fifth arguments for?

Reply:
Those are coming from Nimble parser library traverse sentence, this represent:
(_res, = rest of markup to be parser, args = parser accumulator at point traverse was called, context = a map with specific parameters you wish to create to store some interesting content, _line = line number where excecution is, _offset =  internal offset counter, _last = any_custom_parameters_send_via_traverse_to_internal_function.

https://hexdocs.pm/nimble_parsec/NimbleParsec.html#traverse/3

We are including this names in orer to complement this parameters names.